src/Arcus.BackgroundJobs.Tests.Integration/CloudEvents/CloudEventBackgroundJobTests.cs

```diff
@@ -55,65 +54,87 @@ public async Task CloudEventsBackgroundJobOnNamespace_ReceivesCloudEvents_Proces
             var topicConnectionString = _configuration.GetValue<string>(TopicConnectionStringSecretKey);
             var properties = ServiceBusConnectionStringProperties.Parse(topicConnectionString);
             IEventGridPublisher eventGridPublisher = CreateEventGridPublisher(_configuration);
-            
+
             var options = new WorkerOptions();
             options.Configure(host =>
             {
                 host.ConfigureAppConfiguration(context => context.AddConfiguration(_configuration))
-                    .ConfigureSecretStore((config, stores) => stores.AddConfiguration(config)); 
+                    .ConfigureSecretStore((config, stores) => stores.AddConfiguration(config));
             });
             options.ConfigureLogging(_logger)
                    .AddSingleton(eventGridPublisher)
                    .AddCloudEventBackgroundJob(
                        topicName: properties.EntityPath,
-                       subscriptionNamePrefix: "Test-", 
+                       subscriptionNamePrefix: "Test-",
                        serviceBusNamespaceConnectionStringSecretKey: NamespaceConnectionStringSecretKey)
                    .WithServiceBusMessageHandler<CloudEventToEventGridAzureServiceBusMessageHandler, CloudEvent>();
 
-            CloudEvent expected = CreateCloudEvent();
-            ServiceBusMessage message = CreateServiceBusMessageFor(expected);
-
+            // Act
             await using (var worker = await Worker.StartNewAsync(options))
+            await using (var service = await TestMessagePumpService.StartNewAsync(_configuration, _logger))
             {
-                var producer = TestServiceBusEventProducer.Create(TopicConnectionStringSecretKey, _configuration);
-                await using (var consumer = await TestServiceBusEventConsumer.StartNewAsync(_configuration, _logger))
+                // Assert
+                await service.SimulateCloudEventMessageProcessingAsync(topicConnectionString);
```

Comment:
Hmm, the `Producer`/`Consumer` types in the original tests were introduced to give more insights in how messages are pushed and received. Hiding that functionality again in this `TestMessagePumpService` is maybe taking a step back in visibility. Adding some duplication to increase simplicity is a good thing here. (We have to be accessible for starter devs too).

Reply:
Especially when only a part are written with the test pump. Best to revert this.

Reply:
Reverted this change.